blockchain/hedera.go

```diff
@@ -147,6 +194,75 @@ func (c Client) GetAccountCreditTransactionsAfterTimestamp(accountID string, fro
 	return c.getTransactionsByQuery(transactionsDownloadQuery)
 }
 
+func (c Client) GetAccountByAccountId(accountID string) (*Response, error) {
+	accountQuery := fmt.Sprintf("?account.id=%s", accountID)
+	return c.getAccountByQuery(accountQuery)
+}
+
+func DecodeTransactionMemo(transactionMemo string) ([]byte, error) {
+	return base64.StdEncoding.DecodeString(transactionMemo)
+}
+
+// WaitForTransaction Polls the transaction at intervals.
+func (c Client) WaitForTransaction(accoutId string) {
+
+	logger.Infof("Listening for events on account id: %v", accoutId)
+	lastTransactionTimestamp := time.Now().UnixNano()
+	for {
+		response, err := c.GetAccountCreditTransactionsAfterTimestamp(accoutId, lastTransactionTimestamp)
+
+		if err != nil {
+			logger.Errorf("Error while trying to get account. Error: [%s].", err.Error())
+			return
+		}
+
+		if response != nil {
+			numberOfTransactions := len(response.Transactions)
+			if numberOfTransactions != 0 {
+				transactionTimestamp, err := FromString(response.Transactions[numberOfTransactions-1].ConsensusTimestamp)
+				if err != nil {
+					logger.Errorf(err.Error())
+					return
+				}
+				lastTransactionTimestamp = transactionTimestamp
+			}
+			for _, transaction := range response.Transactions {
+				// This request is targeting a specific jobID
+				decodedMemo, err := DecodeMemo(transaction.MemoBase64)
```

Comment:
Before you decode it, you can check if there is anything - `len(transaction.MemoBase64)`